src/compiler/crystal/util.cr

```diff
@@ -17,26 +13,22 @@ module Crystal
     filename
   end
 
-  def self.error(msg, color, exit_code = 1, stderr = STDERR)
-    stderr.print "Error: ".colorize.toggle(color).red.bold
-    stderr.puts msg.colorize.toggle(color).bright
+  def self.error(msg, exit_code = 1, stderr = STDERR)
+    stderr.print "Error: ".colorize.red.bold
+    stderr.puts msg.colorize.bright
     exit(exit_code) if exit_code
   end
 
   def self.tempfile(basename)
     CacheDir.instance.join("crystal-run-#{basename}.tmp")
   end
 
-  def self.with_line_numbers(source : String, highlight_line_number = nil, color = false)
+  def self.with_line_numbers(source : String, highlight_line_number = nil)
     source.lines.map_with_index do |line, i|
       str = "#{"%4d" % (i + 1)}. #{line.to_s.chomp}"
       target = i + 1 == highlight_line_number
       if target
-        if color
-          str = ">".colorize.green.bold.to_s + str[1..-1].colorize.bold.to_s
-        else
-          str = ">" + str[1..-1]
-        end
+        str = ">".colorize.green.bold.to_s + str.lchop.colorize.bold.to_s
```

Comment:
Instead of concatenating the colorized strings only to collect them in an array and finally join them all, they could just be  appended in a `String.build` directly.
Not necessarily in scope of this PR though.